lib/core/include/termiosUtil.hpp

```diff
@@ -0,0 +1,99 @@
+#ifndef _TERMIOS_UTIL_HPP_
+#define _TERMIOS_UTIL_HPP_
+
+#include <unistd.h>
+#include <errno.h>
+#include <termios.h>
+
+#ifdef __cplusplus
+
+namespace irods
+{
+    // class termiosUtil - at this time, this class encapsulates the work
+    // which has to be done in order to turn echo on or off for a specifc
+    // file desctiptor.  The call can be expanded to encapsulate other
+    // functionality of termios(3).
+
+    class termiosUtil {
+        public:
+            // Class has to be instantiated with a file descriptor.
+            // The default constructor is invalid.
+            explicit termiosUtil(int fd) :
+                valid_(false),
+                errno_copy_(0),
+                fd_(fd),
+                save_c_lflag_(0)
+            {
+                memset(&tty_, 0, sizeof(tty_));
+                if (isatty(fd_))
+                {
+                    if (tcgetattr( fd_, &tty_ ) == 0)
+                    {
+                        save_c_lflag_ = tty_.c_lflag;
+                        valid_ = true;
+                    }
+                    else
+                    {
+                        errno_copy_ = errno;
+                        valid_ = false;
+                    }
+                }
+                else
+                {
+                    errno_copy_ = errno;
+                    valid_ = false;
+                }
+            };
+
+        private:
+            // Invalid constructor.  Ensures instantiation with fd parameter
+            termiosUtil(void) {};
+
+        public:
+            virtual ~termiosUtil(void) = default;
```

Comment:
Is it necessary to declare the destructor as virtual?  Looking over this pull request, I haven't spotted a need for this.  Are you expecting this class to be used as a base?

Reply:
Yes yes and yes.  I don't know that it won't be used as a base class 2 years from now.  (My opinion, which means that this stays).